frontend/components/molecules/SortDropdown/index.tsx

```diff
@@ -31,24 +31,51 @@ const SortDropdown = ({ selectedFilter, filters }: AppProps) => {
                     leaveTo="transform opacity-0 scale-95"
                 >
                     <Menu.Items className="absolute right-0 w-44 origin-top divide-y divide-gray-100 border bg-white drop-shadow-lg">
-                        {filters.map((filter) => {
-                            return (
-                                <div key={filter.id} className="cursor-pointer">
-                                    <Menu.Item>
-                                        {({ active }) => (
-                                            <div
-                                                className={`${
-                                                    active ? 'bg-light-gray' : ''
-                                                } w-full py-2 px-4 text-sm text-gray-900`}
-                                                onClick={filter.onClick}
-                                            >
-                                                {filter.name}
-                                            </div>
-                                        )}
-                                    </Menu.Item>
-                                </div>
-                            )
-                        })}
+                        {!grouped
+                            ? filters.map((filter) => {
+                                  const newFilter = filter as FilterType
+                                  console.log(filter)
+                                  return (
+                                      <div key={newFilter.id} className="cursor-pointer">
+                                          <Menu.Item>
+                                              {({ active }) => (
+                                                  <div
+                                                      className={`${
+                                                          active ? 'bg-light-gray' : ''
+                                                      } w-full py-2 px-4 text-sm text-gray-900`}
+                                                      onClick={newFilter.onClick}
+                                                  >
+                                                      {newFilter.name}
+                                                  </div>
+                                              )}
+                                          </Menu.Item>
+                                      </div>
+                                  )
+                              })
+                            : filters.map((filter, index) => {
```

Comment:
Please create a render function for this one , for it to be more readable and seperate logic in rendering